tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetVarsFromElement.cpp

```diff
@@ -0,0 +1,363 @@
+  }
+  using argument_tags = tmpl::list<Tags::TestSolution>;
+  using base = Square;
+};
+}  // namespace Tags
+
+struct MockComputeTargetPoints {
+  using is_sequential = std::false_type;
+  template <typename Metavariables, typename DbTags>
+  static tnsr::I<DataVector, 3, Frame::Inertial> points(
+      const db::DataBox<DbTags>& /*box*/,
+      const tmpl::type_<Metavariables>& /*meta*/) noexcept {
+    // Need 10 points to agree with test.
+    const size_t num_pts = 10;
+    // Doesn't matter what the points are; they are not used except
+    // that they need to be inside the domain.
+    tnsr::I<DataVector, 3, Frame::Inertial> target_points(num_pts);
+    for (size_t n = 0; n < num_pts; ++n) {
+      for (size_t d=0;d<3;++d) {
+        target_points.get(d)[n] = 1.0 + 0.01 * n + 0.5 * d;
+      }
+    }
+    return target_points;
+  }
+};
+
+struct MockPostInterpolationCallback {
+  template <typename DbTags, typename Metavariables>
+  static void apply(
+      const db::DataBox<DbTags>& box,
+      const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+      const typename Metavariables::temporal_id::type& temporal_id) noexcept {
+    // This callback simply checks that the points are as expected.
+    Slab slab(0.0, 1.0);
+    const TimeStepId first_temporal_id(true, 0, Time(slab, Rational(13, 15)));
+    const TimeStepId second_temporal_id(true, 0, Time(slab, Rational(14, 15)));
+    if (temporal_id == first_temporal_id) {
+      const Scalar<DataVector> expected{
+          DataVector{{0.0, 1.0, 4.0, 9.0, 16.0, 25.0, 36.0, 49.0, 64.0, 81.0}}};
+      CHECK_ITERABLE_APPROX(expected, db::get<Tags::Square>(box));
+    } else if (temporal_id == second_temporal_id) {
+      const Scalar<DataVector> expected{DataVector{
+          {0.0, 4.0, 16.0, 36.0, 64.0, 100.0, 144.0, 196.0, 256.0, 324.0}}};
+      CHECK_ITERABLE_APPROX(expected, db::get<Tags::Square>(box));
+    } else {
+      // Should never get here.
+      CHECK(false);
+    }
+  }
+};
+
+template <typename Metavariables, typename InterpolationTargetTag>
+struct mock_interpolation_target {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using component_being_mocked =
+      intrp::InterpolationTarget<Metavariables, InterpolationTargetTag>;
+  using const_global_cache_tags =
+      tmpl::list<domain::Tags::Domain<Metavariables::volume_dim>>;
+  using simple_tags =
+      db::get_items<typename intrp::Actions::InitializeInterpolationTarget<
+          Metavariables, InterpolationTargetTag>::return_tag_list>;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<
+              simple_tags,
+              typename InterpolationTargetTag::compute_items_on_target>>>,
+      Parallel::PhaseActions<typename Metavariables::Phase,
```

Comment:
you don't need to add empty action lists for phases as long as the full `phase_dependent_action_list` has at least one phase with a non-empty list